src/api/app/models/branch_package/lookup_incident_package.rb

```diff
@@ -0,0 +1,53 @@
+class BranchPackage::LookupIncidentPackage
+  def initialize(params)
+    @package = params[:package]
+    @link_target_project = params[:link_target_project]
+  end
+
+  def package
+    possible_packages = @link_target_project.maintenance_projects.collect do |mp|
+      # only approved maintenance projects
+      next unless maintenance_projects.include?(mp.maintenance_project)
+      # extract possible packages
+      possible_packages(mp.maintenance_project)
+    end.flatten
+    pkg = nil
+    # choose the last one based on the project id in the name openSUSE:Maintenance:1234
+    # can we use the MaintenanceIncident#incident_id ?
+    possible_packages.each do |possible_package|
+      pkg = possible_package if pkg.nil? || possible_package.project.name.gsub(/.*:/, '').to_i > pkg.project.name.gsub(/.*:/, '').to_i
```

Comment:
maybe here instead of gsub of the name, check the `incident_id` ? What do you think @coolo ?